sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/PartitionProcessor.java

```diff
@@ -45,15 +45,20 @@ public void initialize(InitializationContext initializationContext) {
     }
 
     /**
-     * This method is called when a new event is received for this partition. Processing of this event can happen
-     * asynchronously.
+     * This method is called when a new event is received for this partition.
      *
      * @param eventContext The partition information and the next event data from this partition.
      */
     public abstract void processEvent(EventContext eventContext);
 
-    public void processEventBatch(List<EventContext> eventContextBatch) {
-        eventContextBatch.forEach(this::processEvent);
+    /**
+     * This method is called when a batch of events is received for this partition. The default behavior will be to
+     * forward the individual events in this batch to {@link #processEvent(EventContext)}.
+     *
+     * @param eventBatchContext The event batch context containing the batch of events along with partition information.
+     */
+    public void processEventBatch(EventBatchContext eventBatchContext) {
+        throw logger.logExceptionAsError(new UnsupportedOperationException("Processing event batch not implemented"));
```

Comment:
The default behavior mentioned in the JavaDoc does not match the code?

Reply:
Updated the javadoc.